src/libNetwork/P2PComm.cpp

```diff
@@ -422,130 +451,136 @@ void P2PComm::HandleAcceptedConnection(int cli_sock, Peer from)
         return;
     }
 
-    if (startByte == START_BYTE_BROADCAST)
-    {
-        HandleAcceptedConnectionBroadcast(cli_sock, from, messageLength(header),
-                                          move(cli_sock_closer));
-    }
-    else if (startByte == START_BYTE_NORMAL)
-    {
-        HandleAcceptedConnectionNormal(cli_sock, from, messageLength(header),
-                                       move(cli_sock_closer));
-    }
-    else
-    {
-        // Unexpected start byte. Drop this message
-        LOG_GENERAL(WARNING, "Header length or type wrong.");
-    }
-}
-
-void P2PComm::HandleAcceptedConnectionNormal(int cli_sock, Peer from,
-                                             uint32_t message_length,
-                                             SocketCloser cli_sock_closer)
-{
-    vector<unsigned char> message;
+    const uint32_t messageLength = (message[2] << 24) + (message[3] << 16)
+        + (message[4] << 8) + message[5];
 
-    if (!readMsg(&message, cli_sock, from, message_length))
+    // Check for length consistency
+    if (messageLength != message.size() - HDR_LEN)
     {
+        LOG_GENERAL(WARNING, "Incorrect message length.");
         return;
     }
 
-    cli_sock_closer.reset(); // close socket now so it can be reused
-
-    m_dispatcher(message, from);
-}
-
-void P2PComm::HandleAcceptedConnectionBroadcast(int cli_sock, Peer from,
-                                                uint32_t message_length,
-                                                SocketCloser cli_sock_closer)
-{
-    vector<unsigned char> msg_hash;
+    int fd = bufferevent_getfd(bev);
```

Comment:
`bev` is already being freed

Reply:
Thanks, missed that after doing the last revision